pkg/controllers/provisioningv2/rke2/common.go

```diff
@@ -450,3 +460,78 @@ func GetOwnerFromGVK(groupVersion, kind string, obj runtime.Object) (*metav1.Own
 	}
 	return ref, objMeta.GetNamespace(), nil
 }
+
+// CompressInterface is a function that will marshal, gzip, then base64 encode the provided interface.
+func CompressInterface(v interface{}) (string, error) {
+	var b64GZCluster string
+	marshalledCluster, err := json.Marshal(v)
+	if err != nil {
+		return "", err
+	}
+	var b bytes.Buffer
+	gz := gzip.NewWriter(&b)
+	if _, err := gz.Write(marshalledCluster); err != nil {
+		return "", err
+	}
+	if err := gz.Flush(); err != nil {
+		return "", err
+	}
+	if err := gz.Close(); err != nil {
+		return "", err
+	}
+	b64GZCluster = base64.StdEncoding.EncodeToString(b.Bytes())
+	return b64GZCluster, nil
```

Comment:
nit: no need to declare the variable b64GZCluster
```suggestion
	return base64.StdEncoding.EncodeToString(b.Bytes()), nil
```